python/docstring.h

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Neil Vaytet
+#pragma once
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "scipp/dataset/dataset.h"
+#include "scipp/variable/variable.h"
+
+using namespace scipp::variable;
+using namespace scipp::dataset;
+
+template <class T> const std::string type_to_string() {
+  if (std::is_same_v<T, Variable>)
+    return "Variable";
+  if (std::is_same_v<T, VariableView>)
+    return "VariableView";
+  if (std::is_same_v<T, DataArray>)
+    return "DataArray";
+  if (std::is_same_v<T, DataArrayView>)
+    return "DataArrayView";
+  if (std::is_same_v<T, Dataset>)
+    return "Dataset";
+  if (std::is_same_v<T, DatasetView>)
+    return "DatasetView";
+  return "";
+}
+
+class Docstring {
+
+public:
+  Docstring() = default;
+  Docstring(const Docstring &) = default;
+
+  Docstring &description(const std::string &s, const bool append = false);
```

Comment:
```suggestion
public:
  Docstring &description(const std::string &s, const bool append = false);
```
The compiler should generate these. Worse: If you make the copy constructor user-defined (`= default` counts as user-defined), this *prevents* the generation of a move constructor.

You could keep the default constructor in principle, but I don't think it is necessary either?

See also https://en.cppreference.com/w/cpp/language/rule_of_three.

Reply:
Thanks for the link